Modules/xASL_module_ASL.m

```diff
@@ -119,12 +103,40 @@
     x.Q.BackgroundSuppressionNumberPulses = x.BackgroundSuppressionNumberPulses;
 end
 
+%% Define the state names and numbers
+StateName{ 1} = '010_TopUp';
+StateName{ 2} = '020_RealignASL';
+StateName{ 3} = '030_RegisterASL';
+StateName{ 4} = '040_ResampleASL';
+StateName{ 5} = '050_PreparePV';
+StateName{ 6} = '060_ProcessM0';
+StateName{ 7} = '070_CreateAnalysisMask';
+StateName{ 8} = '080_Quantification';
+StateName{ 9} = '090_VisualQC_ASL';
+StateName{10} = '100_WADQC';
 
 %% Change working directory to make sure that unspecified output will go there...
 oldFolder = cd(x.SESSIONDIR);
 
+%% Split ASL and M0 within the ASL time series
+% Run this when the data hasn't been touched yet
+% The first three states are here, because the first two are run only conditionally
+if ~x.mutex.HasState(StateName{1}) && ~x.mutex.HasState(StateName{2}) && ~x.mutex.HasState(StateName{3})
```

Comment:
Come to think of it, we probably need to enforce saving also the first two states even if they were not run, like with the structural module, to keep things the same throughout the pipeline, don't you agree?
If yes, let's make a separate issue for this (and notify Maurice)

Reply:
I think that it is fine to not save states that were not run because of lack of data. But it wouldn't be completely incorrect to save them either. We can ask Maurice if either of those is problematic for him.

I have a slight preference to keep things as they are. The problem can be that if we save states for non-existing data, then if you later on add TOP-UP to an already processed data, it will ignore it if the state will be pre-saved.

Reply:
True, I agree!